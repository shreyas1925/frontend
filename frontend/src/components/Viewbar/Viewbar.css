.viewbar--container{
    width:25%;
    background-color: #272727;
    color:white;
}

.viewbar--header{
    background-color: #5906ce;
    height: 50px;
    padding: 10px;
    line-height: 50px;
    font-size: 25px;
    font-family: "Roboto", sans-serif;
    font-weight: 500;
}

.viewbar--types{
    color:white;
    display: flex;
    font-size: 10px;
    justify-content: space-around;
    margin-top: 40px;
}
.viewbar--contents{
    margin-left: 25px;
    margin-top: -50px;
    cursor: pointer;
}



.viewbar--contents li{
    color:#afafaf;
    font-size: 18px;
    margin-bottom: 15px;
    font-family: "roboto", sans-serif;
}

.viewbar--contents li:hover{
    color: white;
    background-color: black;
    padding:5px;
    transition: all 0.3s ease-in-out;
}  

.viewbar--types h1{
    font-family: "Roboto", sans-serif;
    text-transform: uppercase;
    font-size: 17px;
    color:#afafaf
}

.viewbar--types img{
    width: 35px;
    height: 35px;
    margin-right: 18px;
    margin-top: -6px;
    padding-left:8px ;
}

.builder{
    background-color: black;
    height: 28px;
    line-height: 28px;
}

.builder h2{
    color: white;
    font-size: 15px;
    font-family: "Roboto", sans-serif;
    font-weight: 500;
    margin-left: 25px;
    margin-top: 20px;
    text-transform: uppercase;
}

@media screen and (max-width:900px)
{
    .viewbar--container{
        width:100%;
        background-color: #272727;
        color:white;
    }
}