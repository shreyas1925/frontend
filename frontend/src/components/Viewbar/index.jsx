import React from 'react'
import "./Viewbar.css"
import makeRequest from '../../utils/makeRequest'

import icon from '../../assets/icon.png'
// import ContentTypeContext from '../../context/ContentTypeContext'
const Viewbar = ({contents}) => {

  return (
    <div className='viewbar--container'>
      <div className='viewbar--header'>
        CMS+ 
      </div>
      <div className='viewbar--types'>
          <h1>Collection Types</h1>
          <img src={icon} alt="" />
      </div>
      <div className='viewbar--contents'>
        {
        contents.map((content)=>{
          return (
            <li>{content.name}</li>
          )
        })
        }
      </div>
      <div className='builder'>
        <h2> Content type builder</h2>
      </div>
    </div>
  )
}

export default Viewbar