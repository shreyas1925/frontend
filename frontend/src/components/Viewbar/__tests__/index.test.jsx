import { BrowserRouter } from "react-router-dom";
import Viewbar from "..";
import { render } from "@testing-library/react";
describe("Viewbar", () => {
    const contents = [
        {
            name: "CMS+",
            id: 1,
        },
        {
            name: "CMS+",
            id: 2,
        },
    ]
  it("should render correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Viewbar contents={contents}/>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render and have text", () => {
    const { getByText } = render(
      <BrowserRouter>
        <Viewbar  contents={contents}/>
      </BrowserRouter>
    );
    expect(getByText("Collection Types")).toBeInTheDocument();
  });
});
