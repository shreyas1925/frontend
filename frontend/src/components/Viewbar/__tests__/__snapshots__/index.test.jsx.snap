// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viewbar should render correctly 1`] = `
<DocumentFragment>
  <div
    class="viewbar--container"
  >
    <div
      class="viewbar--header"
    >
      CMS+
    </div>
    <div
      class="viewbar--types"
    >
      <h1>
        Collection Types
      </h1>
      <img
        alt=""
        src="icon.png"
      />
    </div>
    <div
      class="viewbar--contents"
    >
      <li>
        CMS+
      </li>
      <li>
        CMS+
      </li>
    </div>
    <div
      class="builder"
    >
      <h2>
         Content type builder
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
